library/manageiq.py

```diff
@@ -129,24 +161,41 @@ def find_provider_by_name(self, provider_name):
         providers = self.client.collections.providers
         return next((p.id for p in providers if p.name == provider_name), None)
 
-    def add_or_update_provider(self, provider_name, hostname, port, token):
+    def generate_endpoints(self, hostname, port, token, h_hostname, h_port):
+        """ Creates the provider endpoints
+
+        Returns:
+            the provider's endpoints, including the hawkular ones if metrics is required
+        """
+        endpoints = [{'endpoint': {'role': 'default', 'hostname': hostname,
+                                   'port': port},
+                      'authentication': {'role': 'bearer', 'auth_key': token}}]
+
+        # add hawkular endpoints if metrics is True
+        if self.module.params['metrics']:
```

Comment:
@dkorn can `metrics` be missing? If you use `self.module.params.get('metrics')`, then you would receive `None` in case `metrics` is not defined.